framework/src/userobject/SolutionUserObject.C

```diff
@@ -633,6 +633,51 @@ SolutionUserObject::pointValue(Real t, const Point & p, const unsigned int local
   return val;
 }
 
+RealGradient
+SolutionUserObject::pointValueGradient(Real t, const Point & p, const std::string & var_name) const
+{
+  const unsigned int local_var_index = getLocalVarIndex(var_name);
+  return pointValueGradient(t, p, local_var_index);
+}
+
+RealGradient
+SolutionUserObject::pointValueGradient(Real t, const Point & p, const unsigned int local_var_index) const
+{
+  // Create copy of point
+  Point pt(p);
```

Comment:
You could just pass the Point in by value if you have to make a copy of it anyway.